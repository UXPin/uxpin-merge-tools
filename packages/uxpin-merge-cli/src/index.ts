import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
