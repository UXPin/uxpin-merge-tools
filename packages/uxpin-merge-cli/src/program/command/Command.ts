export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
