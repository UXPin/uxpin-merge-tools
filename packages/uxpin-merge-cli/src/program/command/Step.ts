import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
