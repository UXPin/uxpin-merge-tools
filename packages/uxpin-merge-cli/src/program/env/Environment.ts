export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
