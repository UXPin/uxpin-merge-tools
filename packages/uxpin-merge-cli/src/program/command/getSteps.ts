import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
