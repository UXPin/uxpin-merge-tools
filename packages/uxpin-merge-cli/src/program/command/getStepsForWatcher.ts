import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
