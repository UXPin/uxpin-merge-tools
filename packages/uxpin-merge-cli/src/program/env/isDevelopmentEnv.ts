import { Environment } from './Environment';

export function isDevelopmentEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.DEVELOPMENT;
}
