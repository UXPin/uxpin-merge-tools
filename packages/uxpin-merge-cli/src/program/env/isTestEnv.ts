import { Environment } from './Environment';

export function isTestEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.TEST;
}
