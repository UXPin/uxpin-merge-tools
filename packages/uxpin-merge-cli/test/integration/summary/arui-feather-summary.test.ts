import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { aruiFeatherSummaryStub } from '../../resources/stubs/aruiFeather';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT:number = 30000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('summary command integration', () => {
  const { getTlsPort } = setupStubbyServer(aruiFeatherSummaryStub);

  describe('summary command prints ', () => {
    it('prints the list of components found in arui-feather example', () => {
      // when
      return runUXPinMergeCommand({
        cwd: 'resources/repos/arui-feather',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [
          Command.SUMMARY,
          '--config="../../configs/arui-feather-uxpin.config.js"',
        ],
      }).then((output) => {
        // then
        expect(output).toMatchSnapshot();
      });
    });
  });
});
