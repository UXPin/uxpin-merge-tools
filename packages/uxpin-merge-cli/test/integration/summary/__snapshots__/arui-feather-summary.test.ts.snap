// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary command integration summary command prints  prints the list of components found in arui-feather example 1`] = `
"[1mUncategorized[22m

    [1mAmount[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mAttach[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mButton[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCalendar[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCalendarInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCardInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCheckBox[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCheckBoxGroup[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mCollapse[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mDropdown[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mEmailInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mFlagIcon[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mForm[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mFormField[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mHeading[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIcon[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIconButton[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mInputAutocomplete[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mInputGroup[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIntlPhoneInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mIsolatedContainer[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mLabel[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mLink[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mList[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mMaskedInput[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mMenu[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mMenuItem[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mMoneyInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mMq[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mNotification[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mParagraph[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mPhoneInput[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mPlate[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mPopup[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mPopupContainerProvider[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mPopupHeader[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mProgressBar[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mRadio[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mRadioGroup[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mRenderInContainer[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mResizeSensor[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSelect[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSidebar[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSlideDown[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSpin[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mSwipeable[22m
        ðŸ“œ documentation: [31mâœ˜[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTabItem[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTabs[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTagButton[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mTextarea[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [32mâœ”[39m
        ðŸŽ›  presets: [31mâœ˜[39m

    [1mThemeProvider[22m
        ðŸ“œ documentation: [32mâœ”[39m
        ðŸ’¡ examples: [31mâœ˜[39m
        ðŸŽ›  presets: [31mâœ˜[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/icon-button/icon-button.jsx[39m

[33mwarning [39mCannot parse type of a property 'size' in:
[34msrc/intl-phone-input/intl-phone-input.jsx[39m

[33mwarning [39mCannot parse type of a property 'value' in:
[34msrc/intl-phone-input/intl-phone-input.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/swipeable/swipeable.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/tab-item/tab-item.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/tabs/tabs.jsx[39m

[33mwarning [39mCannot serialize component properties in:
[34msrc/tag-button/tag-button.jsx[39m

"
`;
